test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs

```diff
@@ -231,6 +232,51 @@ public void TestGetMsbuildDirectoryFromPATHENV()
             }
         }
 
+        [Theory]
+        [InlineData('"')]
+        [InlineData('<')]
+        [InlineData('>')]
+        [InlineData('|')]
+        public void GetMsbuildDirectoryFromPath_PATHENVWithInvalidChars_Succeeds(char invalidChar)
+        {
+            using (var vsPath = TestDirectory.Create())
+            {
+
+                if (RuntimeEnvironmentHelper.IsMono)
+                { // Mono does not have SxS installations so it's not relevant to get msbuild from the path.
+                    return;
```

Comment:
Do the mono check before running `TestDirectory.Create()`, so we don't waste time creating and deleting directories never used.